### ISSUE TITLE
[java] FinalFieldCouldBeStatic: Constant in @interface incorrectly reported as "could be made static"

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#275]
**Rule Set:** design.html#FinalFieldCouldBeStatic

**Code Sample demonstrating the issue:**

```java
public @interface MetricType {
    int CHECKBOX = 0, COUNTER = 1, SPINNER = 2, NOTE = 3, STOPWATCH = 4, HEADER = 5;
}
```

-->  but all fields in interfaces/@interfaces are static and final by default

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
Issue confirmed, PR submitted. Thanks for the report!
